# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsIFrame.cpp
# Commit: 4fca9cae3f42
# Full Hash: 4fca9cae3f42010acd2f985f54aa969d3e7fff5e
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2023-01-04 04:29:41
# Description:
#   Bug 1807769 - Null-check frame selection in nsIFrame::HasSelectionInSubtree r=emilio
#   
#   Be more careful when accessing the frame selection, which might be
#   nullptr.
#   
# ==============================================================================

diff -r 0910d52020ec -r 4fca9cae3f42 layout/generic/nsIFrame.cpp
--- a/layout/generic/nsIFrame.cpp	Tue Jan 03 11:16:11 2023 +0000
+++ b/layout/generic/nsIFrame.cpp	Tue Jan 03 11:41:09 2023 +0000
@@ -6971,6 +6971,10 @@
   }
 
   RefPtr<nsFrameSelection> frameSelection = GetFrameSelection();
+  if (!frameSelection) {
+    return false;
+  }
+
   const Selection* selection =
       frameSelection->GetSelection(SelectionType::eNormal);
   if (!selection) {